vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.xray.exporter;
+
+import static org.apache.commons.lang3.Validate.notEmpty;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.vividus.bdd.model.jbehave.Meta;
+import org.vividus.bdd.model.jbehave.Scenario;
+import org.vividus.bdd.model.jbehave.Story;
+import org.vividus.jira.JiraFacade;
+import org.vividus.xray.configuration.XrayExporterOptions;
+import org.vividus.xray.facade.TestCaseParameters;
+import org.vividus.xray.facade.XrayFacade;
+import org.vividus.xray.reader.JsonResourceReader;
+import org.vividus.xray.reader.JsonResourceReader.FileEntry;
+
+@Component
+public class XrayExporter
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(XrayExporter.class);
+
+    private static final String SEMICOLON = ";";
+
+    @Autowired private XrayExporterOptions xrayExporterOptions;
+    @Autowired private XrayFacade xrayFacade;
+    @Autowired private JiraFacade jiraFacade;
+
+    public void exportResults() throws IOException
+    {
+        for (Story story : readStories())
+        {
+            for (Scenario scenario : story.getScenarios())
+            {
+                String storyTitle = story.getPath();
+                String scenarioTitle = scenario.getTitle();
+
+                List<Meta> scenarioMeta = scenario.getMeta();
+                if (isSkipped(scenarioMeta))
+                {
+                    LOGGER.atInfo()
+                          .addArgument(scenarioTitle)
+                          .addArgument(storyTitle)
+                          .log("Skip export of {} scenario in {} story");
+                    continue;
+                }
+
+                TestCaseParameters parameters = new TestCaseParameters();
+                parameters.setLabels(getMetaValues(scenarioMeta, "xray.labels"));
+                parameters.setComponents(getMetaValues(scenarioMeta, "xray.components"));
+                parameters.setSummary(scenarioTitle);
+                parameters.setStory(storyTitle);
+                parameters.setSteps(scenario.collectSteps());
+
+                String testCaseId = getMetaValuesStream(scenarioMeta, "testCaseId").findFirst().orElse(null);
```

Comment:
open question: do we need to support mapping of multiple test cases to single scenario

Reply:
as to test cases we expect to support 1 to 1 mapping, as to requirements, we ok with 1 to 1 at the moment, but in the future we will expand it